packages/cli/src/commands/deploy.ts

```diff
@@ -69,104 +72,78 @@ async function run(options: DeployCommandOptions): Promise<void> {
-    ...new Set(Object.values(deployManifest.stages).map((d) => d.package)),
+    ...new Set(allStepsFromAllSequences.map((step) => step.package)),
   ];
 
   sanitizePackages(packageNames);
 
   await project.cacheDeployModules(packageNames);
 
   const packageMap: Record<string, DeployPackage> = {};
-  const stageToPackageMap: Record<string, DeployPackage> = {};
 
   for await (const packageName of packageNames) {
     packageMap[packageName] = await project.getDeployModule(packageName);
   }
 
-  Object.entries(deployManifest.stages).forEach(([stageName, stageValue]) => {
-    stageToPackageMap[stageName] = packageMap[stageValue.package];
-  });
+  const stepToPackageMap: Record<
+    string,
+    DeployPackage & { sequenceName: string }
+  > = {};
 
-  validateManifestWithExts(deployManifest, stageToPackageMap);
+  for (const step of allStepsFromAllSequences) {
+    stepToPackageMap[step.name] = {
+      ...packageMap[step.package],
+      sequenceName: step.sequenceName,
+    };
+  }
 
-  const handlers: Record<string, DeployerHandler> = {};
-  const roots: { handler: DeployerHandler; uri: Uri }[] = [];
+  validateManifestWithExts(deployManifest, stepToPackageMap);
 
-  // Create all handlers
-  Object.entries(deployManifest.stages).forEach(([stageName, stageValue]) => {
-    const publisher = stageToPackageMap[stageName].deployer;
-    handlers[stageName] = new DeployerHandler(
-      stageName,
-      publisher,
-      stageValue.config,
-      print
-    );
-  });
-
-  // Establish dependency chains
-  Object.entries(deployManifest.stages).forEach(([key, value]) => {
-    const thisHandler = handlers[key];
-
-    if (value.depends_on) {
-      // Depends on another stage
-      handlers[value.depends_on].addNext(thisHandler);
-    } else if (value.uri) {
-      // It is a root node
-      roots.push({ uri: new Uri(value.uri), handler: thisHandler });
-    } else {
-      throw new Error(
-        `Stage '${key}' needs either previous (depends_on) stage or URI`
-      );
-    }
+  const sequences = deployManifest.sequences.map((sequence) => {
+    const steps = sequence.steps.map((step) => {
+      return new Step({
+        name: step.name,
+        uriOrStepResult: step.uri,
+        deployer: stepToPackageMap[step.name].deployer,
+        config: step.config ?? {},
+      });
+    });
+
+    return new Sequence({
+      name: sequence.name,
+      steps,
+      config: sequence.config ?? {},
+      printer: print,
+    });
   });
 
-  // Execute roots
-
-  const resultLists: ResultList[] = [];
+  const sequenceResults: SequenceResult[] = [];
 
-  for await (const root of roots) {
-    print.info(`\nExecuting deployment chain: \n`);
-    root.handler.getDependencyTree().printTree();
-    print.info("");
-    await root.handler.handle(root.uri);
-    resultLists.push(root.handler.getResultsList());
+  for await (const sequence of sequences) {
+    const sequenceResult = await sequence.run();
+    sequenceResults.push(sequenceResult);
```

Comment:
Do sequence need to be run in sequence? Why don't we parallelize it?

Reply:
Yep, done